src/common/client.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""EtcdClient utility class to connect to etcd server and execute commands with etcdctl."""
+
+import json
+import logging
+import subprocess
+
+logger = logging.getLogger(__name__)
+
+
+class EtcdClient:
+    """Handle etcd client connections and run etcdctl commands."""
+
+    def __init__(
+        self,
+        client_url: str,
+    ):
+        self.client_url = client_url
+
+    def get_endpoint_status(self) -> dict:
+        """Run the `endpoint status` command and return the result as dict."""
+        endpoint_status = {}
+        if result := self._run_etcdctl(
+            command="endpoint",
+            subcommand="status",
+            endpoints=self.client_url,
+            output_format="json",
+        ):
+            endpoint_status = json.loads(result)[0]
+
+        return endpoint_status
+
+    def _run_etcdctl(
+        self,
+        command: str,
+        subcommand: str | None,
+        endpoints: str,
+        output_format: str | None = "simple",
```

Comment:
1. what does `output_format = None` represent? 
2. should the default not be always `json` to simplify the usage of output, so that `json.loads` happens only once in this method?

Reply:
1. This means you don't have to set `output_format` when calling `_run_etcdctl()` if you only need `simple` output. Only if you need `json`, you need to set it.
2. I assume the majority of our calls will only require `simple` output format, especially for the imperative commands like `member add`, `user add`, `role add` and so on. Only the `list` or `status` commands will require json-output. That's why I opted for the default to be `simple`.